Libraries/DirectMLX.h

```diff
@@ -4291,6 +4291,191 @@ namespace dml
 
         return output;
     }
+
+    struct MultiHeadAttentionOutputs
+    {
+        Expression output;
+        Optional<Expression> outputPresentKey;
+        Optional<Expression> outputPresentValue;
+    };
+
+    inline MultiHeadAttentionOutputs MultiHeadAttention(
+        Optional<Expression> query,
+        Optional<Expression> key,
+        Optional<Expression> value,
+        Optional<Expression> stackedQueryKey,
+        Optional<Expression> stackedKeyValue,
+        Optional<Expression> stackedQueryKeyValue,
+        Optional<Expression> bias,
+        Optional<Expression> mask,
+        Optional<Expression> relativePositionBias,
+        Optional<Expression> pastKey,
+        Optional<Expression> pastValue,
+        Optional<Expression> pastSequenceLengths,
+        float scale,
+        float maskFilterValue,
+        uint32_t queryHeadCount,
+        uint32_t keyValueHeadCount,
+        DML_MULTIHEAD_ATTENTION_MASK_TYPE maskType,
+        bool computeOutputPresentKeyValue,
+        Optional<uint32_t> maxSequenceLength = {})
+    {
+        assert(query || stackedQueryKey || stackedQueryKeyValue);
+
+        detail::GraphBuilder* builder = nullptr;
+
+        if (query)
+        {
```

Comment:
(minor 🤷) `assert(stackedKeyValue || (key && value));` for assertion consistency with the other branches?